site/src/components/RichParameterInput/RichParameterInput.tsx

```diff
@@ -0,0 +1,205 @@
+        </span>
+        <span className={styles.labelDescription}>{parameter.description}</span>
+      </label>
+    )
+  }
+
+  return (
+    <label htmlFor={parameter.name}>
+      <span className={styles.labelDescription}>{parameter.name}</span>
+    </label>
+  )
+}
+
+export interface RichParameterInputProps {
+  disabled?: boolean
+  parameter: TemplateVersionParameter
+  onChange: (value: string) => void
+}
+
+export const RichParameterInput: FC<RichParameterInputProps> = ({
+  disabled,
+  onChange,
+  parameter,
+  ...props
+}) => {
+  const styles = useStyles()
+
+  return (
+    <Stack direction="column" spacing={0.75}>
+      <ParameterLabel parameter={parameter} />
+      <div className={styles.input}>
+        <RichParameterField
+          {...props}
+          disabled={disabled}
+          onChange={onChange}
+          parameter={parameter}
+        />
+      </div>
+    </Stack>
+  )
+}
+
+const RichParameterField: React.FC<RichParameterInputProps> = ({
+  disabled,
+  onChange,
+  parameter,
+  ...props
+}) => {
+  const styles = useStyles();
+
+  if (isBoolean(parameter)) {
+    return (
+      <RadioGroup
+        defaultValue={parameter.default_value}
+        onChange={(event) => {
+          onChange(event.target.value)
+        }}
+      >
+        <FormControlLabel
+          disabled={disabled}
+          value="true"
+          control={<Radio color="primary" size="small" disableRipple />}
+          label="True"
+        />
+        <FormControlLabel
+          disabled={disabled}
+          value="false"
+          control={<Radio color="primary" size="small" disableRipple />}
+          label="False"
+        />
+      </RadioGroup>
+    )
+  }
+
+  if (parameter.options.length > 0) {
+    return (
+      <RadioGroup
+        defaultValue={parameter.options[0].value}
+        onChange={(event) => {
+          onChange(event.target.value)
+        }}
+      >
+        {
+          parameter.options.map((option) => (
+            <FormControlLabel
+              disabled={disabled}
+              key={option.name}
+              value={option.value}
+              control={<Radio color="primary" size="small" disableRipple />}
+              label={(
+                <span>
+                  {option.icon && (
+                    <img
+                      className={styles.optionIcon}
+                      alt="Parameter icon"
+                      src={option.icon}
+                      style={{
+                        pointerEvents: "none",
+                      }}
+                    />
+                  )}
+                  {option.name}
+                </span>
+              )}
+            />
+          ))
+        }
+      </RadioGroup>
+    )
+  }
+
+  // A text field can technically handle all cases!
+  // As other cases become more prominent (like filtering for numbers),
+  // we should break this out into more finely scoped input fields.
+  return (
+    <TextField
+      {...props}
```

Comment:
@BrunoQuaresma Could you please take a look at this PR?

It looks like I messed up some component properties or incorrectly manage the state. I admit that I got stuck for a few hours on this.

```
Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
    at input
    at div
    at InputBase2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-CC2W5OQU.js?v=aabcc073:326:30)
    at WithStyles2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-WTKBUNCY.js?v=aabcc073:3404:35)
    at OutlinedInput2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-CC2W5OQU.js?v=aabcc073:742:23)
    at WithStyles2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-WTKBUNCY.js?v=aabcc073:3404:35)
    at div
    at FormControl2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-HUDXJYUM.js?v=aabcc073:73:24)
    at WithStyles2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-WTKBUNCY.js?v=aabcc073:3404:35)
    at TextField2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-VM5J5TEI.js?v=aabcc073:183:28)
    at WithStyles2 (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-WTKBUNCY.js?v=aabcc073:3404:35)
    at RichParameterField (http://127.0.0.1:8080/src/components/RichParameterInput/RichParameterInput.tsx?t=1674665405142:145:3)
    at div
    at div
    at Stack (http://127.0.0.1:8080/src/components/Stack/Stack.tsx:46:3)
    at RichParameterInput (http://127.0.0.1:8080/src/components/RichParameterInput/RichParameterInput.tsx?t=1674665405142:101:3)
    at div
    at Stack (http://127.0.0.1:8080/src/components/Stack/Stack.tsx:46:3)
    at div
    at div
    at Stack (http://127.0.0.1:8080/src/components/Stack/Stack.tsx:46:3)
    at form
    at main
    at div
    at Margins (http://127.0.0.1:8080/src/components/Margins/Margins.tsx:36:3)
    at FullPageHorizontalForm (http://127.0.0.1:8080/src/components/FullPageForm/FullPageHorizontalForm.tsx:22:3)
    at CreateWorkspacePageView (http://127.0.0.1:8080/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx?t=1674665499946:44:7)
    at CreateWorkspacePage (http://127.0.0.1:8080/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx?t=1674665499946:28:26)
    at RenderedRoute (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2402:5)
    at RenderedRoute (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2402:5)
    at RenderedRoute (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2402:5)
    at Outlet (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2641:26)
    at Suspense
    at div
    at div
    at DashboardProvider (http://127.0.0.1:8080/src/components/Dashboard/DashboardProvider.tsx:26:3)
    at DashboardLayout (http://127.0.0.1:8080/src/components/Dashboard/DashboardLayout.tsx:32:18)
    at RenderedRoute (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2402:5)
    at Outlet (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2641:26)
    at RequireAuth (http://127.0.0.1:8080/src/components/RequireAuth/RequireAuth.tsx:24:23)
    at RenderedRoute (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2402:5)
    at Routes (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2701:5)
    at Router (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-IUIN2ZYE.js?v=aabcc073:2648:15)
    at BrowserRouter (http://127.0.0.1:8080/node_modules/.vite/deps/react-router-dom.js?v=b0592f46:220:5)
    at Suspense
    at AppRouter
    at AuthProvider (http://127.0.0.1:8080/src/components/AuthProvider/AuthProvider.tsx:22:3)
    at ErrorBoundary (http://127.0.0.1:8080/src/components/ErrorBoundary/ErrorBoundary.tsx:7:5)
    at ThemeProvider (http://127.0.0.1:8080/node_modules/.vite/deps/chunk-536WFZRT.js?v=aabcc073:515:24)
    at r2 (http://127.0.0.1:8080/node_modules/.vite/deps/react-helmet-async.js?v=236d04da:417:21)
    at App
```